File: backend/models/child.model.js
const mongoose = require("mongoose");

const ChildSchema = new mongoose.Schema(
  {
    name: { type: String, required: [true, "Name is required"], trim: true },
    gender: {
      type: String,
      enum: ["Boy", "Girl"],
      required: [true, "Gender is required"],
    },
    birthDate: { type: Date, required: [true, "Birth date is required"] },
    heightAtBirth: { type: Number, min: [0, "Height must be positive"] },
    weightAtBirth: { type: Number, min: [0, "Weight must be positive"] },
    headCircumferenceAtBirth: {
      type: Number,
      min: [0, "Head circumference must be positive"],
    },
    bloodType: {
      type: String,
      match: [/^(A|B|AB|O)[+-]$/, "Invalid blood type"],
    },
    parentId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    // deviceId: { type: String, required: false, unique: true },
    photo: {
      type: String,
      default: "uploads/child-default.jpg",
      validate: {
        validator: function (value) {
          return !value || /\.(jpg|jpeg|png|gif)$/i.test(value);
        },
        message: "Photo must be a valid image file",
      },
    },
    favorite: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Doctor",
        required: false,
      },
    ],
  },
  { timestamps: true }
);

module.exports = mongoose.model("Child", ChildSchema);

File: lib/models/child_model.dart
import 'package:flutter/material.dart';

class Child {
  final String id;
  final String name;
  final String gender;
  final DateTime birthDate;
  final double heightAtBirth;
  final double weightAtBirth;
  final double headCircumferenceAtBirth; // الحقل الجديد
  final String bloodType;
  final String? photo;
  final String? parentPhone;

  Child({
    required this.id,
    required this.name,
    required this.gender,
    required this.birthDate,
    required this.heightAtBirth,
    required this.weightAtBirth,
    required this.headCircumferenceAtBirth, // الحقل الجديد
    required this.bloodType,
    this.photo,
    this.parentPhone,
  });

  factory Child.fromJson(Map<String, dynamic> json) {
    return Child(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      gender: json['gender'] ?? 'Male',
      birthDate: DateTime.parse(json['birthDate'] ?? DateTime.now().toIso8601String()),
      heightAtBirth: (json['heightAtBirth'] ?? 0).toDouble(),
      weightAtBirth: (json['weightAtBirth'] ?? 0).toDouble(),
      headCircumferenceAtBirth: (json['headCircumferenceAtBirth'] ?? 0).toDouble(), // الحقل الجديد
      bloodType: json['bloodType'] ?? '',
      photo: json['photo'],
      parentPhone: json['parentPhone'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'gender': gender,
      'birthDate': birthDate.toIso8601String(),
      'heightAtBirth': heightAtBirth,
      'weightAtBirth': weightAtBirth,
      'headCircumferenceAtBirth': headCircumferenceAtBirth, // الحقل الجديد
      'bloodType': bloodType,
      'photo': photo,
      'parentPhone': parentPhone,
    };
  }
}